File: device-sdk/src/main/java/io/oddworks/device/metric/OddVideoStopMetric.java
package io.oddworks.device.metric;

import android.util.Log;

import org.json.JSONException;
import org.json.JSONObject;

public class OddVideoStopMetric extends OddMetric {
    private static final String TAG = OddVideoStopMetric.class.getSimpleName();

    protected static String videoStopAction = ACTION_VIDEO_STOP;
    protected static boolean videoStopEnabled = false;

    private int mElapsed;
    private int mDuration;

    public OddVideoStopMetric() {

    }

    public static void setAction(String action) {
        videoStopAction = action;
    }

    public static void setEnabled(boolean enabled) {
        videoStopEnabled = enabled;
    }


    public OddVideoStopMetric setElapsed(int elapsed) {
        mElapsed = elapsed;
        return this;
    }

    public OddVideoStopMetric setDuration(int duration) {
        mDuration = duration;
        return this;
    }

    @Override
    public String getAction() {
        return videoStopAction;
    }

    @Override
    public boolean getEnabled() {
        return videoStopEnabled;
    }

    @Override
    public JSONObject toJSONObject() {
        JSONObject json = super.toJSONObject();

        try {
            JSONObject attributes = json.getJSONObject("attributes");

            attributes.put("elapsed", mElapsed);
            attributes.put("duration", mDuration);

            json.put("attributes", attributes);
        } catch (JSONException e) {
            Log.d(TAG, e.toString());
        }

        return json;
    }
}

File: device-sdk/src/main/java/io/oddworks/device/metric/OddVideoStopMetric.kt
package io.oddworks.device.metric

import android.util.Log

import org.json.JSONException
import org.json.JSONObject

class OddVideoStopMetric(contentType: String, contentId: String, title: String, meta: JSONObject? = null, val elapsed: Int = 0, val duration: Int = 0) : OddMetric(contentType, contentId, title, meta) {

    override val action: String
        get() = OddVideoStopMetric.action

    override val enabled: Boolean
        get() = OddVideoStopMetric.enabled

    override fun toJSONObject(): JSONObject {
        val json = super.toJSONObject()

        try {
            val data = json.getJSONObject("data")
            val attributes = data.getJSONObject("attributes")

            attributes.put("elapsed", elapsed)
            attributes.put("duration", duration)

            data.put("attributes", attributes)
            json.put("data", data)
        } catch (e: JSONException) {
            Log.d(TAG, e.toString())
        }

        return json
    }

    override fun toString(): String {
        return "$TAG (type=$type, contentType=$contentType, contentId=$contentId, meta=${meta.toString()}, elapsed=$elapsed, duration=$duration)"
    }

    companion object {
        private val TAG = OddVideoStopMetric::class.java.simpleName

        var action = OddMetric.ACTION_VIDEO_STOP
        var enabled = false
    }
}
